### PR TITLE
Fixing Hedgehog abilities hurting dead, and related

### PR DESCRIPTION
**Pull request by Kiara-Miller**
## About The Pull Request
Should fix the issues in https://github.com/cmss13-devs/cmss13/issues/358
Base Runner and Hedgehog abilities have had some changes on their underlying code, that may impact how they work in edge cases.
Introduces a new bullet flag, AMMO_XENO_BONE in order to properly facilitate the fixes.
Fixes Xeno bone based abilities being able to hurt corpses if used while standing on them.
Fixes Xeno bone based abilities being able to hurt corpses if used on nearby vertical corpses.
Fixes Xeno bone based abilities from form being able to hurt caps.
Fixes shrapnel bypassing some checks, if created on top of a mob. Notably these checks will now happen when the 'shrapnel' is caused by Xeno abilities too. There may a small performance hit, this needs testing.
No significant balance impact is expected, tho in a few cases the above abilities may deal a bit less damage due to the bug fixes.

## Why It's Good For The Game
Bugfixes and standardizes bone based abilities into their own thing when it comes to bullet flags as they don't neatly fit into  either acid or toxic.

## Changelog

:cl:ito726
fix: Base Runner and Hedgehog Ravager form being able to hurt corpses and caps.
/:cl:

### CODE REVIEW

**Comment by Carnage068** [comment]
salt pr. just remove hedgehog from the fucking game at this point lmao

**Comment by usnpeepoo** [comment]
> salt pr. just remove hedgehog from the fucking game at this point lmao. if you want to nerf it that's fine but don't fucking lie about it being a direct nerf, come on.

Hey, just so you know you're meant to leave feedback in the CM discord related to this PR. No longer allowed to leave feedback in the GitHub page/PR page.

File: code/modules/projectiles/ammo_datums.dm [hunk 1]
```diff
@@ -2707,6 +2707,10 @@
 	shrapnel_chance = 60
 
 /datum/ammo/xeno/bone_chips/on_hit_mob(mob/M, obj/item/projectile/P)
+	if(iscarbon(M))
+		var/mob/living/carbon/C = M
+		if(C.status_flags & XENO_HOST && HAS_TRAIT(C, TRAIT_NESTED) || C.stat == DEAD)
```
Comment (Geevies) [inline code/modules/projectiles/ammo_datums.dm:2712]:
```suggestion
		if((HAS_FLAG(C.status_flags, XENO_HOST) && HAS_TRAIT(C, TRAIT_NESTED)) || C.stat == DEAD)
```

File: code/modules/projectiles/ammo_datums.dm [hunk 2]
```diff
@@ -2733,6 +2737,10 @@
     shrapnel_chance = 0
 
 /datum/ammo/xeno/bone_chips/spread/runner/on_hit_mob(mob/M, obj/item/projectile/P)
+    if(iscarbon(M))
+        var/mob/living/carbon/C = M
+        if(C.status_flags & XENO_HOST && HAS_TRAIT(C, TRAIT_NESTED) || C.stat == DEAD)
```
Comment (Geevies) [inline code/modules/projectiles/ammo_datums.dm:2742]:
```suggestion
        if((HAS_FLAG(C.status_flags, XENO_HOST) && HAS_TRAIT(C, TRAIT_NESTED)) || C.stat == DEAD)
```

**Review by Geevies — CHANGES_REQUESTED** [review]

**Review by Geevies — APPROVED** [review]

**Pull request merged**